### ISSUE TITLE
MaterialArrayLayout throws error if childData is undefined.

### ISSUE THREAD

**Issue by shivgarg5676** [eclipsesource/jsonforms#1375]
I have a very nested schema. 
```
{
   "type":"object",
   "properties":{
      "componentTitle":{
         "type":"string"
      },
      "showOnNavigation":{
         "type":"boolean"
      },
      "domIdentifier":{
         "type":"string"
      },
      "description":{
         "type":"string"
      },
      "modules":{
         "type":"array",
         "items":{
            "type":"object",
            "properties":{
               "title":{
                  "type":"string"
               },
               "meta":{
                  "type":"array",
                  "items":{
                     "type":"object",
                     "properties":{
                        "count":{
                           "type":"number"
                        },
                        "description":{
                           "type":"string"
                        },
                        "iconSlug":{
                           "type":"string"
                        }
                     },
                     "required":[
                        "count",
                        "description",
                        "iconSlug"
                     ]
                  }
               },
               "instructors":{
                  "type":"array",
                  "items":{
                     "type":"object",
                     "properties":{
                        "name":{
                           "type":"string"
                        },
                        "imgURL":{
                           "type":"string"
                        }
                     },
                     "required":[
                        "name",
                        "imgURL"
                     ]
                  }
               },
               "subModules":{
                  "type":"array",
                  "items":{
                     "type":"object",
                     "properties":{
                        "title":{
                           "type":"string"
                        },
                        "type":{
                           "type":"string"
                        },
                        "duration":{
                           "type":"number"
                        }
                     },
                     "required":[
                        "title",
                        "type",
                        "duration"
                     ]
                  }
               },
               "tools":{
                  "type":"array",
                  "items":{
                     "type":"object",
                     "properties":{
                        "name":{
                           "type":"string"
                        },
                        "iconSlug":{
                           "type":"string"
                        }
                     },
                     "required":[
                        "name",
                        "iconSlug"
                     ]
                  }
               }
            },
            "required":[
               "title",
               "meta",
               "instructors"
            ]
         }
      },
      "summary":{
         "type":"object",
         "properties":{
            "items":{
               "type":"array",
               "items":{
                  "type":"object",
                  "properties":{
                     "count":{
                        "type":"string"
                     },
                     "description":{
                        "type":"string"
                     }
                  },
                  "required":[
                     "count",
                     "description"
                  ]
               }
            }
         },
         "required":[
            "items"
         ]
      }
   },
   "required":[
      "componentTitle",
      "showOnNavigation",
      "domIdentifier",
      "description",
      "modules",
      "summary"
   ]
}
```
The problem is when i try to remove element from the **`modules array`** it throws the following error.
```
Cannot read property 'title' of undefined
    at Function.push../node_modules/@jsonforms/material-renderers/lib/layouts/MaterialArrayLayout.js.exports.mapStateToExpandPanelProps [as mapToProps]
```
![image](https://user-images.githubusercontent.com/6873680/58029818-00e7f800-7b3b-11e9-8bd2-aeed0eb8e64e.png)
**This does not occuer everytime. but consistanty occurs if i try to delete last element.**

I tried looking into the Source code of `MaterialArrayLayout.tsx`. The problem is that at `line number 225` component data is getting resolved into undefined and due to that `line number 226` breaks. 
![image](https://user-images.githubusercontent.com/6873680/58029743-d433e080-7b3a-11e9-846d-5d0212ae6358.png)

Is there a need to have `undefined` check or there is something else going on?

**Comment by edgarmueller** [comment]
Thanks for the report, we'll try to have a look soon.

**Comment by shivgarg5676** [comment]
#1377

**Comment by shivgarg5676** [comment]
CodeSandbox url 
https://codesandbox.io/embed/jsonformsreactseed-1qyxt

**Comment by edgarmueller** [comment]
I've had a look at this and it definitely is an error on our side that occurs in the 2.2.3 release, but I can not reproduce it with the current master nor with 6580c02c45fec40c51dd86b195ce02b5be1fb8df. I think it's likely we'll accept your contribution (and thanks for the PR!), but it'd be interesting to find out why it didn't seem to happen in our dev environment. @eneufeld will have a look.

**Comment by edgarmueller** [comment]
So both me and @eneufeld had a look and we came to the conclusion that this issue is related to a wrong flow of events/some kind of race condition when pressing the delete button as it actually triggers two things at once: the deletion of the element and the expansion of the panel. This is why it seems as if the element is first deleted, but the expansion panel component also re-renders simultaneously, which must not be the case. The good news is that you already fix this issue  with #1384 where you've added `event.stopPropagation` within `removeItems`, which seems to fix the issue. As mentioned, we were not able to reproduce this issue in the dev environment (we still don't know why exactly) but we were able to verify the fix by altering sources within the seed. I'm closing this issue and #1377 therefore. If the problem should persist for some reason, please feel free to re-open.
